ecosystem/indexer-grpc/indexer-transaction-generator/src/config.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, HashSet},
+    path::{Path, PathBuf},
+};
+use url::Url;
+
+const IMPORTED_TRANSACTIONS_FOLDER: &str = "imported_transactions";
+
+#[derive(Parser)]
+pub struct IndexerCliArgs {
+    /// Path to the configuration file with `TransactionGeneratorConfig`.
+    #[clap(long)]
+    pub config: PathBuf,
+
+    /// Path to the output folder where the generated transactions will be saved.
+    #[clap(long)]
+    pub output_folder: PathBuf,
+}
+
+impl IndexerCliArgs {
+    pub async fn run(&self) -> anyhow::Result<()> {
+        // Read the configuration file.
+        let config_raw = tokio::fs::read_to_string(&self.config)
+            .await
+            .with_context(|| format!("Failed to read configuration file: {:?}", self.config))?;
+
+        // Parse the configuration.
+        let config: TransactionGeneratorConfig = serde_yaml::from_str(&config_raw)
+            .with_context(|| format!("Failed to parse configuration file: {:?}", self.config))?;
+
+        // Run the transaction generator.
+        config.run(&self.output_folder).await
+    }
+}
+
+/// Overall configuration for the transaction generator.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TransactionGeneratorConfig {
+    pub import_config: TransactionImporterConfig, // TODO: Add scripted transaction generation configuration.
+}
+
+impl TransactionGeneratorConfig {
+    pub async fn run(&self, output_path: &Path) -> anyhow::Result<()> {
+        let import_config_path = output_path.join(IMPORTED_TRANSACTIONS_FOLDER);
+        // Check if the output folder exists.
+        if !import_config_path.exists() {
+            tokio::fs::create_dir_all(&import_config_path).await?;
+        }
+        self.import_config.run(&import_config_path).await
+    }
+}
+
+/// Configuration for importing transactions from multiple networks.
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct TransactionImporterConfig {
+    pub mainnet_config: Option<TransactionImporterPerNetworkConfig>,
+    pub testnet_config: Option<TransactionImporterPerNetworkConfig>,
+    pub devnet_config: Option<TransactionImporterPerNetworkConfig>,
+    pub localnet_config: Option<TransactionImporterPerNetworkConfig>,
```

Comment:
Instead of hardcoding all the networks, the config could take in an arbitrary map of network configs, wdyt? This could support more custom networks, like moonnet, etc.

Reply:
done.